src/main/java/com/spotify/annoy/ANNIndex.java

```diff
@@ -14,8 +14,8 @@
 public class ANNIndex implements AnnoyIndex {
 
 
-  private final ArrayList<Integer> roots;
-  private MappedByteBuffer annBuf;
+  private final ArrayList<Long> roots;
+  private Map<Integer, MappedByteBuffer> buffers = new HashMap<>();
```

Comment:
This could just be an ArrayList, right? Or even a MappedByteBuffer[] since you know how many there are to begin with?

Hash lookups are much more expensive, and you don't need to do them.